### PR TITLE
[dv] initial icache testbench

### PR DESCRIPTION
**Pull request by udinator**
The initial commit of the testbench for the ibex_icache module generated by the `uvmdvgen` tool.
The testbench files and .hjson supporting files for `dvsim` tool are all contained within `dv/uvm/icache/[data/doc/dv]`.

There are several things to note with this PR:
1) A directory `vendor/lowrisc_ip` has been created, which contains the `dv_base` testbench library, the `dv_utils` package, and various common tools, including the `uvmdvgen` tool for auto-generating testbenches and most importantly, the `dvsim` tool by which to actually simulate this icache testbench. For the time being, these tools have been manually copied over from the lowRISC/opentitan repository until they are spun off into a separate repo themselves. These tools and utilities from lowRISC/opentitan are the bulk of this PR and should not warrant a review.
2) Several fusesoc .core files have been created in the existing Ibex testbench environment and agent directories, these are required since the icache testbench will use the currently existing ibex memory interface agent. These core files are dual purpose, as they will also be used when the Ibex testbenches switches over to using the `dvsim` .hjson based flow.
3) To run builds and simulations, the `dvsim` tool must be invoked from the `ibex/` top level directory.

This PR provides all of the testbench components and run flow utilities needed to compile the icache testbench and run tests once they have been implemented.

Signed-off-by: Udi <udij@google.com>

### CODE REVIEW

File: dv/uvm/core_ibex/common/ibex_mem_intf_agent/ibex_mem_intf_agent_pkg.sv [hunk 1]
```diff
@@ -2,7 +2,6 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
-`include "ibex_mem_intf.sv"
```
Comment (udinator) [inline dv/uvm/core_ibex/common/ibex_mem_intf_agent/ibex_mem_intf_agent_pkg.sv]: @taoliug @sriyerg - this will be a potential pain point, as if I remove this inclusion the main Ibex testbench will not compile, but if I keep it as is, the icache testbench will not compile since it uses a different simulation flow instead of `dvsim`. Once Ibex is integrated with `dvsim` (which will happen by the end of the week), this issue can temporarily be avoided, but will still remain an issue for any users simulating Ibex with the existing simulation flow using the `Makefile` and `sim.py`.

Comment (sriyerg) [reply dv/uvm/core_ibex/common/ibex_mem_intf_agent/ibex_mem_intf_agent_pkg.sv]: I assume `sim.py` uses a static filelist. Why not add the `*_intf.sv` to the .f file (right before this package)?

File: dv/uvm/icache/data/common_sim_cfg.hjson [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    {
+      name: bar
+      build_opts: ["+define+bbaru1",
+                   "+define+bbaru2",
+                   "+define+bbaru3"]
+      run_opts: ["+rbar1u=1",
+                 "+rbar2u=2",
+                 "+rbar3u=3"]
+    }
+  ]
```
Comment (sriyerg) [inline dv/uvm/icache/data/common_sim_cfg.hjson]: You can get rid of this - this is just an example of how you can set 'build_modes'.

File: dv/uvm/icache/data/common_sim_cfg.hjson [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  sw_build_dir:       ""
+  sw_root_dir:        ""
+
+  // pass and fail patterns
+  pass_patterns:    ["^TEST PASSED (UVM_)?CHECKS$"]
+  fail_patterns:    ["^UVM_ERROR\\s[^:].*$",
+                     "^Assert failed: ",
+                     "^\\s*Offending '.*'",
+                     "^TEST FAILED (UVM_)?CHECKS$"]
+
```
Comment (sriyerg) [inline dv/uvm/icache/data/common_sim_cfg.hjson]: Reminder to update this as needed for Ibex.

Comment (udinator) [reply dv/uvm/icache/data/common_sim_cfg.hjson]: Yep, those modifications will be coming in a later PR if needed.

**Comment by imphil** [comment]
Please use the vendor tool if you copy over code from OpenTitan. Do one commit where you add the vendor.hjson file, and another one after you run the tool to keep the auto-generated changes and the import separate and easier to review.

File: dv/uvm/icache/data/common_modes.hjson [hunk 4]
```diff
@@ -0,0 +1,38 @@
+// Copyright lowRISC contributors.
```
Comment (sriyerg) [inline dv/uvm/icache/data/common_modes.hjson]: Consider moving these common hjson files to a more common area the Ibex testbench can point to as well.

File: dv/uvm/icache/data/vcs/cover.cfg [hunk 5]
```diff
@@ -0,0 +1,9 @@
++tree tb.dut
```
Comment (sriyerg) [inline dv/uvm/icache/data/vcs/cover.cfg]: Update this as needed as well, later.

File: dv/uvm/icache/data/vcs/vcs.mk [hunk 6]
```diff
@@ -0,0 +1,117 @@
+# Copyright lowRISC contributors.
```
Comment (sriyerg) [inline dv/uvm/icache/data/vcs/vcs.mk]: Remove this file - it is not needed. It s a vestige from the legacy Mekefile flow.

File: vendor/lowrisc_ip/common_ifs/clk_if.sv [hunk 7]
```diff
@@ -0,0 +1,29 @@
+// Copyright lowRISC contributors.
```
Comment (sriyerg) [inline vendor/lowrisc_ip/common_ifs/clk_if.sv]: Consider removing this and using the more feature rich one below instead.

**Comment by udinator** [comment]
> It's not really a manageable situation to have two copies of code around with no automated way to update it between repositories. Please use the vendoring tool, or let me know how it fails to be useful here.

@imphil, that's a good point, probably using the vendoring tool would be the best solution. My plan is this: I will strip the utility/tool code directories from this PR and only leave the icache testbench files and any related .hjson/.core files. In parallel, I'll open up some other PRs to get the .vendor.hjson files in and then vendor in the relevant tool directories from opentitan.
How does this sound?

Looping in @rswarbrick just as a FYI.

**Comment by imphil** [comment]
Thanks @udinator, sounds good! (You can also do it in separate commits as part of this PR, if you prefer.)

File: dv/uvm/core_ibex/common/ibex_mem_intf_agent/ibex_mem_intf_agent_pkg.sv (hunk 1 above)
Comment (udinator) [reply dv/uvm/core_ibex/common/ibex_mem_intf_agent/ibex_mem_intf_agent_pkg.sv]: That would probably work, I'll try that.

File: vendor/lowrisc_ip/common_ifs/clk_if.sv (hunk 7 above)
Comment (udinator) [reply vendor/lowrisc_ip/common_ifs/clk_if.sv]: Does this still apply if I am vendoring in the common_ifs/ directory?

File: dv/uvm/icache/data/common_modes.hjson (hunk 4 above)
Comment (udinator) [reply dv/uvm/icache/data/common_modes.hjson]: Yep, I'm thinking about how to structure the shared directories

**Comment by udinator** [comment]
@imphil - sure, that is a good option here, I'll split this PR into several commits and then merge.

File: dv/uvm/icache/data/vcs/cover.cfg (hunk 5 above)
Comment (udinator) [reply dv/uvm/icache/data/vcs/cover.cfg]: Yep, this cover.cfg file can pretty much be used verbatim for Ibex as well.

**Comment by udinator** [comment]
Update: The several PRs for modifications to the tools meant to be vendored in are blocking as of right now, so instead of waiting for those, I'll simply just push the icache testbench code, and then follow up with some PRs for the vendored code and hjson files.

**Comment by rswarbrick** [comment]
@udinator I'm not convinced that's the right way to do it. Am I right in thinking that the testbench code is impossible to build without these vendored files? If you won't have time for the PRs you mentioned as "follow-ups" here, could you just list the files that you think are needed and maybe I could make those patches tomorrow?

**Comment by imphil** [comment]
Just to confirm: this new code won't be usable after being merged, but it also won't break existing code? If that's the case, then we can merge it from my PoV to make your life easier juggling around various pull requests.

File: vendor/lowrisc_ip/common_ifs/clk_if.sv (hunk 7 above)
Comment (sriyerg) [reply vendor/lowrisc_ip/common_ifs/clk_if.sv]: Aah, no.

**Comment by udinator** [comment]
@imphil - we actually just got the necessary PRs in the main opentitan repo merged, so I'll actually just go ahead and vendor in the code in this PR itself and submit.

File: ibex_icache.core [hunk 8]
```diff
@@ -0,0 +1,19 @@
+CAPI=2:
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+name: "lowrisc:ip:ibex_icache:0.1"
```
Comment (imphil) [inline ibex_icache.core]: The IP cores in the ibex repository are in the `lowrisc:ibex` namespace.

File: ibex_icache.core [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+      - shared/rtl/ram_1p.sv
+      - rtl/ibex_icache.sv
+    file_type: systemVerilogSource
+
+targets:
+  default: &default_target
+    filesets:
+      - files_rtl
+    toplevel: ibex_icache
+    default_tool: vcs
```
Comment (imphil) [inline ibex_icache.core]: Why is this file necessary? The icache is already included in the `ibex_core.core` file.

File: dv/uvm/icache/dv/ibex_icache_sim.core [hunk 10]
```diff
@@ -0,0 +1,25 @@
+CAPI=2:
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+name: "lowrisc:dv:ibex_icache_sim:0.1"
```
Comment (imphil) [inline dv/uvm/icache/dv/ibex_icache_sim.core]: Please put this into the `lowrisc:ibex` namespace.

**Comment by udinator** [comment]
> Just to better understand what's going on in `[dv] initial icache testbench`:
> 
> * This adds initial infrastructure and the testbench? All the tool configuration files etc. will be needed only once? (It's a massive amount of code!)

Yup, the tool configuration files (vcs/excelium) are needed to run the dvsim tool, and of course, all of the code in the vendor/lowrisc_ip/ directory is generally needed just to run the dvsim tool as well.

> * This testbench is not integrated into the existing testbench which uses riscv-dv?

No, the testbench for the icache is going to be a completely standalone testbench that solely tests `rtl/ibex_icache.sv` and as such doesn't use the riscv-dv generator. The only commonality between the icache testbench and the testbench for the entire Ibex core is the memory interface UVM agent, which both utilize to communicate with external memory.

> * All of this also needs documentation in https://ibex-core.readthedocs.io/en/latest/verification.html telling us how to run the thing.

With regards to documentation, I didn't add any in this PR just because the testbench is still in the very early stages right now (it can only be compiled, and cannot run any tests due to none actually existing).
@rswarbrick will need to fill out the DV plan documentation and .hjson testplan (located in `dv/uvm/icache/doc` and `dv/uvm/icache/data` respectively), and then parts of the DV documentation can be copied over to the main Ibex verification documentation.
Is there anything you wanted me to add to the existing verification documentation for the time being? If so, I can either add another commit to this PR or open a separate PR for it.

**Comment by imphil** [comment]
Re docs: Please add least add a line or two how to run the simulation, even if it doesn't do anything yet (or link to existing documentation, if that exists).

File: vendor/lowrisc_ip/dv_utils/dv_utils.core [hunk 11]
```diff
@@ -10,8 +10,8 @@ filesets:
     depend:
       - lowrisc:dv:common_ifs
       - lowrisc:prim:assert:0.1
-      - lowrisc:constants:top_pkg
```
[the remark this answers is not in the document]
Comment (udinator) [reply vendor/lowrisc_ip/dv_utils/dv_utils.core]: Ahh you're right, I'll patch it through the vendor tool.

Yes, you're also right that the DV utils shouldn't depend on top_pkg regardless, this is something that will be fixed in the future, and will be re-vendored into Ibex when that happens.

**Comment by udinator** [comment]
Re docs: Sure, I'll add a few lines about that, with some pointers to dvsim and icache documentation and testplan.

File: dv/uvm/icache/data/vcs/vcs_fsdb.tcl [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+# +Reg_Only                  Dumps only reg type signals
+# +IO_Only                   Dumps only IO port signals
+# +by_file=<filename>        File to specify objects to add
+# +all                       Dumps memories, MDA signals, structs, unions,power, and packed structs
+
+if {[info exists ::env(WAVES)]} {
+  if {$::env(WAVES) == 1} {
+    fsdbDumpfile  $::env(DUMP_FILE)
+    fsdbDumpvars  0 $::env(TB_TOP) +all
+    fsdbDumpSVA   0 $::env(TB_TOP)
```
[the remark this answers is not in the document]
Comment (udinator) [reply dv/uvm/icache/data/vcs/vcs_fsdb.tcl]: Good point, I'll add .vpd support as in #630

File: ibex_icache.core (hunk 8 above)
Comment (udinator) [reply ibex_icache.core]: Thanks for the pointer, will update this accordingly.

File: ibex_icache.core (hunk 9 above)
Comment (udinator) [reply ibex_icache.core]: I wanted to create a core file just for the `ibex_icache.sv`design since the `ibex_core.core` file includes all of the other RTL code, which isn't needed for the icache testbench.

File: dv/uvm/icache/dv/ibex_icache_sim.core (hunk 10 above)
Comment (udinator) [reply dv/uvm/icache/dv/ibex_icache_sim.core]: Will do.

File: ibex_icache.core (hunk 9 above)
Comment (imphil) [reply ibex_icache.core]: In this case, please remove the cache-related files from the main core file, and depend on this file instead.

Comment (udinator) [reply ibex_icache.core]: Sure, will do

**Comment by imphil** [comment]
I'd prefer to have a *single* top-level directory for generated files, which is `build` in our case. (The .gitignore file in OpenTitan shows clearly what happens if you don't do that.)

However, given that this is effectively a port of code from OT, I'd be OK with having a separate scratch directory here as well. But let's not take this then as a "free for all" signal to add even more build output files in the source tree.

**Comment by rswarbrick** [comment]
Actually, I think we can control the scratch directory with the `-sr` argument to `dvsim.py`. I ran a canned command that @udinator gave me (out of band) without that argument, and generated stuff in the `scratch` directory. I guess we just need to set up our scripts to make sure it gets passed properly. Sorry for the noise.

**Comment by udinator** [comment]
@imphil, @rswarbrick - the scratch/ directory is the default output directory created by the dvsim tool, you can use the -sr cmdline option to specify an output directory. I'll add the -sr option to the compile command used in the brief documentation, that way we don't have to add more stuff to the .gitignore .

File: dv/uvm/icache/doc/ibex_icache_dv_plan.md [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+---
+
+<!-- Copy this file to hw/ip/ibex_icache/doc/ibex_icache_dv_plan.md and make changes as needed.
+For convenience 'ibex_icache' in the document can be searched and replaced easily with the
+desired IP (with case sensitivity!). Also, use the testbench block diagram
+located at OpenTitan team drive / 'design verification'
+as a starting point and modify it to reflect your ibex_icache testbench and save it
+to hw/ip/ibex_icache/doc/tb.svg. It should get linked and rendered under the block
+diagram section below. Please update / modify / remove sections below as
+applicable. Once done, remove this comment before making a PR. -->
```
Comment (rswarbrick) [inline dv/uvm/icache/doc/ibex_icache_dv_plan.md]: Since you've presumably done the find/replace, this comment needs deleting! (See last sentence :-))

File: doc/verification.rst [hunk 14]
```diff
@@ -203,12 +206,58 @@ The entirety of this flow is controlled by the Makefile found at
...
+This testbench is located at `dv/uvm/icache/dv
+<https://github.com/lowRISC/ibex/blob/master/dv/uvm/icache/dv>`_.
+
+As Icache verification is being carried out as part of the OpenTitan open-source
+project, the testbench derives from the `dv_lib UVM class library
+<https://github.com/lowRISC/opentitan/tree/master/hw/dv/sv/dv_lib>`_, which is a set of extended UVM
+classes that provides basic UVM testbench functionality and components.
+
+This DV environment will be compiled and simulated using the `dvsim simulation tool
+<https://github.com/lowRISC/opentitan/tree/master/util/dvsim>`_, the master
```
[the remark this answers is not in the document]
Comment (udinator) [reply doc/verification.rst]: Done

File: doc/verification.rst [hunk 15]
```diff
@@ -203,12 +206,58 @@ The entirety of this flow is controlled by the Makefile found at
...
+As this testbench is still in its infancy, it is currently only able to be compiled, as no tests or
+sequences are implemented, nor are there any entries in the testplan file.
+To build the testbench locally using the VCS simulator, ensure that the ``SCRATCH_ROOT``
+environment variable is set to the intended output directory, and then run
+the following command from the root of the Ibex repository:
+
+.. code-block:: bash
+
+   ./vendor/lowrisc_ip/dvsim/dvsim.py dv/uvm/icache/dv/ibex_icache_sim_cfg.hjson --build-only
+   --skip-ral --purge
```
[the remark this answers is not in the document]
Comment (udinator) [reply doc/verification.rst]: Sure, will do

File: dv/uvm/icache/doc/ibex_icache_dv_plan.md (hunk 13 above)
Comment (udinator) [reply dv/uvm/icache/doc/ibex_icache_dv_plan.md]: Done

**Comment by rswarbrick** [comment]
@udinator The memory interface for the I-side of the machine is different from the D-side (which is what you put in the common directory): it's even got different ports. The D-side stuff should probably be left in the `core_ibex` directory.

**Comment by udinator** [comment]
> @udinator The memory interface for the I-side of the machine is different from the D-side (which is what you put in the common directory): it's even got different ports. The D-side stuff should probably be left in the `core_ibex` directory.

If I'm understanding your concern correctly, that is what I already do. The only interface agent provided in dv/uvm/icache/dv is for the interface between the icache and the rest of the Ibex fetch stage. The memory interface between the icache and any external memory model already exists inside of core_ibex/, I have not moved that into the icache/dv directory.

**Comment by rswarbrick** [comment]
Hi Udi. Sorry, I think I got confused. The icache is only used on the I side of the machine (obviously). The interface between the icache and the core will be modelled by the stuff you've put in `ibex_icache_agent`, as you say. The interface between the icache and the instruction bus is different from the interface between the D side of the machine and the memory bus and in fact I'm going to model it with a separate (still icache-specific) agent.

I'd mistakenly thought that commit `2431551` ("Reorganize ibex dv files") was part of this PR, but it's not (it came from #618).

So I stand by my claim that the `ibex_mem_intf_slave_agent` stuff won't help with icache verification, but I now realise that it's not relevant to this PR! Sorry :-)

**Comment by udinator** [comment]
Hi Rupert, could you elaborate a bit why the `ibex_mem_intf_slave_agent` won't be helpful for icache verification? That should be the agent through which the icache sends any instruction fetch requests to the external memory model, and the `ibex_mem_intf_slave_agent` is already fully implemented to act as both the D-side and I-side interface agent for the top ibex testbench, and since the external-facing icache interface is the same as the external interface of the prefetch buffer (see `core_ibex/tb/core_ibex_tb_top.sv` for the interface connections) this agent should be perfectly valid to use here, especially since it would save you the time/effort of re-implementing all of the logic.

File: vendor/lowrisc_ip/dv_utils/dv_utils.core [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+name: "lowrisc:dv:dv_utils"
+description: "DV utilities"
+
+filesets:
+  files_dv:
+    depend:
+      - lowrisc:dv:common_ifs
+      - lowrisc:prim:assert:0.1
+    files:
+      - top_pkg.sv
```
Comment (imphil) [inline vendor/lowrisc_ip/dv_utils/dv_utils.core]: Has this file been modified? That's the file at the given revision in opentitan: https://github.com/lowRISC/opentitan/blob/c383de06/hw/dv/sv/dv_utils/dv_utils.core

If you want to do modifications you need to add a patch and setup patching in `vendor/lowrisc_dv_utils.lock.hjson`. You can find documentation for that process at https://docs.opentitan.org/doc/ug/vendor_hw/#how-to-add-patches-on-top-of-the-imported-code.

In the same patch, also add the top_pkg.sv file that you currently add as separate file into the `vendor/lowrisc_ip/dv_utils/` directory. If you don't add `top_pkg.sv` as part of a patch it will be overwritten with the next run of the vendor tool.

(That being said, the easier option might be to add a core named `lowrisc:constants:top_pkg` without patching anything.)

**Comment by rswarbrick** [comment]
@udinator, @imphil This pull request has sprawled a bit and it's quite hard to see what's left to do before it can land. Here's what I think is left. Philipp, please jump in if I've missed something.

For this PR (just one thing!):

 * Please squash "basic icache test documentation" with "initial icache testbench": there's no need to keep these separate.

For a later PR:

 * [Update pass / fail patterns](#discussion_r393403286) (@rswarbrick can do this)

For next time (not worth changing this now, I don't think)

 * It's better if a patch set starts by installing dependencies and then uses them. So the vendoring stuff would come before "initial icache testbench". That way, a future reader doesn't have to look into the future from a commit to work out what's going on.
 * Lots of your commits include stuff unrelated to the commit message. For example "add vendor .hjson files for dv tools" includes changes to the namespacing in `ibex_icache.core`. These namespacing changes look sensible, but have nothing to do with hjson files: I think they probably just got squashed into the wrong commit.

Finally, about the UVM agents, let's park this conversation until I've pushed up a candidate DV plan (which I'll be able to do as soon as this PR lands).

**Comment by udinator** [comment]
> @udinator, @imphil This pull request has sprawled a bit and it's quite hard to see what's left to do before it can land. Here's what I think is left. Philipp, please jump in if I've missed something.
> 
> For this PR (just one thing!):
> 
> * Please squash "basic icache test documentation" with "initial icache testbench": there's no need to keep these separate.

Sure.

> * Please could you bump the version of the vendored dvsim code from OpenTitan? A couple of bugfixes have landed in the last few days and we may as well start with something newer.

Will do.

> For a later PR:
> 
> * [Update pass / fail patterns](#discussion_r393403286) (@rswarbrick can do this)

I'll handle this one actually, I'm working on a PR to integrate Ibex with the dvsim tool, this includes some changes that I've already made to this file.
 
> For next time (not worth changing this now, I don't think)
> 
> * It's better if a patch set starts by installing dependencies and then uses them. So the vendoring stuff would come before "initial icache testbench". That way, a future reader doesn't have to look into the future from a commit to work out what's going on.
> * Lots of your commits include stuff unrelated to the commit message. For example "add vendor .hjson files for dv tools" includes changes to the namespacing in `ibex_icache.core`. These namespacing changes look sensible, but have nothing to do with hjson files: I think they probably just got squashed into the wrong commit.

Sure, I'll rearrange the commits (won't be that much effort) to make things a bit more logical in terms of how the commits build on each other.

> Finally, about the UVM agents, let's park this conversation until I've pushed up a candidate DV plan (which I'll be able to do as soon as this PR lands).

Agreed.

File: vendor/lowrisc_ip/dv_utils/dv_utils.core (hunk 16 above)
Comment (udinator) [reply vendor/lowrisc_ip/dv_utils/dv_utils.core]: Sure, I'll add the top_pkg core file to icache/dv/ to prevent having to patch anything.

**Comment by udinator** [comment]
@rswarbrick @imphil - I restructured the commits in this PR to start with adding the relevant vendor files, then vendoring in all of the tools, and then the main icache testbench commit, and some other small modifications on top of that. The commit messages are much more clear this time around and there is a better separation between the changes in each commit, so is more modular as well.
As per @rswarbrick suggestion, documentation has been integrated into the `initial icache testbench` commit to clear things up even more.
Also, all of the vendor tools have been "refreshed", so they now reflect the most up-to-date version of the tools in lowRISC/opentitan.

**Comment by imphil** [comment]
Ci fails with 

```
verilator -f lowrisc_ibex_ibex_core_tracing_0.1.vc -Wall --unroll-count 72
%Error: ../../../rtl/ibex_tracer_pkg.sv:7: syntax error, unexpected IDENTIFIER, expecting PACKAGE-IDENTIFIER or STRING
import ibex_pkg::*;
       ^~~~~~~~
%Error: ../../../rtl/ibex_tracer_pkg.sv:114: syntax error, unexpected endpackage
endpackage
^~~~~~~~~~
%Error: ../../../rtl/ibex_tracer.sv:74: syntax error, unexpected IDENTIFIER, expecting PACKAGE-IDENTIFIER or STRING
  import ibex_pkg::*;
         ^~~~~~~~
%Error: Exiting due to 3 error(s)
```


The `lowrisc:ibex:ibex_tracer` actually needs to depend on a package containing ibex_pkg.sv. Maybe it's best to split this package out into a separate core, and then depend on it in the tracer and in the core itself. If you don't see the solution right away, I can have a look later today.

**Comment by imphil** [comment]
Hm, thinking about it, there might be a simpler fix. Let's see how CI copes with https://github.com/lowRISC/ibex/pull/737.

**Comment by udinator** [comment]
> This looks good! Let's get #737 in and then rebase on top of it, and then CI should be fine.

Sounds good!
I'll wait for a last look-over from @rswarbrick and then will merge!

**Comment by imphil** [comment]
@udinator I've found another place where we needed to fix the core files; I've fixed that in #739, can you rebase this PR on top of master?
